src/Ocelot/Multiplexer/MultiplexingMiddleware.cs

```diff
@@ -255,5 +255,27 @@ protected virtual Task MapAsync(HttpContext httpContext, Route route, List<HttpC
 
         var aggregator = _factory.Get(route);
         return aggregator.Aggregate(route, httpContext, contexts);
+    }
+
+    protected virtual async Task<Stream> CloneRequestBodyAsync(HttpContext source)
+    {
+        source.Request.EnableBuffering();
+        if (source.Request.Body.Position == 0)
+        {
+            var targetBuffer = new MemoryStream();
+            if (source.Request.ContentLength is not null)

```

Comment:
What to do if the header is `null`?

Reply:
I just forward an empty `MemoryStream`.

Default implementation uses a `NullStream`, an internal inaccessible subclass.

Reply:
This information should be included in the feature documentation. Developers need to understand that if a request contains a body, the `Content-Length` header must be present; otherwise, the body will be transmitted as `null` or empty.
Do you agree?

Reply:
Hum that's actually an HTTP convention `https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Length`

I don't think it's necessary to document this: not sending a content-length header will prevent all http compliant servers not to read the body (in theory).